src/engines/nwn2/roster.cpp

```diff
@@ -39,6 +39,42 @@ Roster::Roster() {
 Roster::~Roster() {
 }
 
+/** Return the roster name of the first member. */
+Common::UString Roster::getFirstRosterMember() {
+	Common::UString name = "";
+	_lastRetrieved = 0;
+
+	// Set to the first member's name
+	if (!_members.empty())
+		name = (_members.begin())->rosterName;
+
+	return name;
+}
+
+/** Return the roster name of the next member. */
+Common::UString Roster::getNextRosterMember() {
+	Common::UString name = "";
+
+	if (_members.empty()) {
+		_lastRetrieved = 0;
+	} else if (_lastRetrieved < _members.size()) {
+		uint32 index = 0;
+
+		// Find the matching member
+		for (std::list<Member>::const_iterator it = _members.begin(); it != _members.end(); ++it, index++) {
```

Comment:
You could just use `std::advance()` on an iterator returned from `_members.begin()` here.

Alternatively, `_lastRetrieved` could just be a `std::list<Member>::const_iterator` itself. `_members.end()` would be a good guard-value then.